### PR TITLE
Folder iterator performance optimization

### PR DESCRIPTION
**Pull request by k8n**
See [this comment](https://github.com/sookasa/box.py/commit/8f59c68e3b60d949b5eab78917d4030946df9815#commitcomment-5748894)

I'd like to say that a test is forthcoming :-)

### CODE REVIEW

**Comment by tals** [comment]
Your code has a bug - if content['entries'] is empty, it will be stuck in a loop.

**Comment by k8n** [comment]
Sorry, tals. Drive-by coding failed on my part. Hopefully I got it right this time.

File: box/client.py [hunk 1]
```diff
@@ -418,7 +418,10 @@ def get_folder_iterator(self, folder_id):
         while content['entries']:
             for entry in content['entries']:
                 yield entry
-
+             
+            if  offset + len(content['entries']) >= content['total_count']:
```
Comment (tals) [inline box/client.py:423]: nitpicking, but - can you put a comment there to explain the condition&break?
It makes parsing through code more pleasant :)

If not I can just add it later

File: tests/test_client.py [hunk 2]
```diff
@@ -228,16 +228,43 @@ def test_get_folder_iterator(self):
...
 
         (flexmock(client)
             .should_receive('get_folder_content')
             .with_args(666, limit=1000, offset=1000)
-            .and_return({'entries': None})
+            .and_return({'entries': [1000], 'total_count': 1001})
             .once())
 
-        self.assertListEqual(list(client.get_folder_iterator(666)), list(range(10)))
+        self.assertListEqual(list(client.get_folder_iterator(666)), list(range(1001)))
```
Comment (tals) [inline tests/test_client.py]: switch to assertSequenceEqual() instead of forcing lists

**Comment by k8n** [comment]
Cleaned up. Rolled-up changes into a single commit. Added requested changes. (Not going to try patch software working off an iPad in bed anymore :-). Thanks for your guidance.

**Comment by k8n** [comment]
Merge me? I want to kill my fork.

**Pull request merged**